components/clp-package-utils/clp_package_utils/scripts/native/search.py

```diff
@@ -162,33 +171,82 @@ async def do_search(
-        run_function_in_process(
-            create_and_monitor_job_in_db,
-            db_config,
-            results_cache,
-            wildcard_query,
-            tags,
-            begin_timestamp,
-            end_timestamp,
-            ignore_case,
-            max_num_results,
-            path_filter,
-            do_count_aggregation,
-            count_by_time_bucket_size,
+    if do_count_aggregation is None:
+        host = None
+        for ip in set(socket.gethostbyname_ex(socket.gethostname())[2]):
+            host = ip
+            break
+        if host is None:
+            logger.error("Could not determine IP of local machine.")
+            return -1
+        try:
+            server = await asyncio.start_server(
+                client_connected_cb=worker_connection_handler,
+                host=host,
+                port=0,
+                family=socket.AF_INET,
+            )
+        except asyncio.CancelledError:
+            # Search cancelled
+            return
+        port = int(server.sockets[0].getsockname()[1])
+        server_task = asyncio.ensure_future(server.serve_forever())
+
+        db_monitor_task = asyncio.ensure_future(
+            run_function_in_process(
+                create_and_monitor_job_in_db,
+                db_config,
+                results_cache,
+                wildcard_query,
+                tags,
+                begin_timestamp,
+                end_timestamp,
+                ignore_case,
+                path_filter,
+                (host, port),
+                do_count_aggregation,
+            )
         )
-    )
 
-    # Wait for the job to complete or an error to occur
-    try:
-        await db_monitor_task
-    except asyncio.CancelledError:
-        pass
+        # Wait for the job to complete or an error to occur
+        pending = [server_task, db_monitor_task]
+        try:
+            done, pending = await asyncio.wait(pending, return_when=asyncio.FIRST_COMPLETED)
+            if db_monitor_task in done:
+                server.close()
+                await server.wait_closed()
+            else:
+                logger.error("server task unexpectedly returned")
+                db_monitor_task.cancel()
+                await db_monitor_task
+        except asyncio.CancelledError:
+            server.close()
+            await server.wait_closed()
+            await db_monitor_task
+    else:
+        db_monitor_task = asyncio.ensure_future(
+            run_function_in_process(
+                create_and_monitor_job_in_db,
+                db_config,
+                results_cache,
+                wildcard_query,
+                tags,
+                begin_timestamp,
+                end_timestamp,
+                ignore_case,
+                path_filter,
+                None,
+                do_count_aggregation,
+            )
+        )
+        try:
+            await db_monitor_task
+        except asyncio.CancelledError:
+            pass
```

Comment:
I don't think we need to swallow the error since `asyncio.run` does that for us.